common/src/main/java/org/opensearch/ml/common/model/MLGuard.java

```diff
@@ -0,0 +1,161 @@
+    private Client client;
+
+    public MLGuard(Guardrails guardrails, NamedXContentRegistry xContentRegistry, Client client) {
+        this.xContentRegistry = xContentRegistry;
+        this.client = client;
+        if (guardrails == null) {
+            return;
+        }
+        Guardrail inputGuardrail = guardrails.getInputGuardrail();
+        Guardrail outputGuardrail = guardrails.getOutputGuardrail();
+        if (inputGuardrail != null) {
+            fillStopWordsToMap(inputGuardrail, stopWordsIndicesInput);
+            inputRegex = inputGuardrail.getRegex() == null ? new ArrayList<>() : Arrays.asList(inputGuardrail.getRegex());
+            inputRegexPattern = inputRegex.stream().map(reg -> Pattern.compile(reg)).collect(Collectors.toList());
+        }
+        if (outputGuardrail != null) {
+            fillStopWordsToMap(outputGuardrail, stopWordsIndicesOutput);
+            outputRegex = outputGuardrail.getRegex() == null ? new ArrayList<>() : Arrays.asList(outputGuardrail.getRegex());
+            outputRegexPattern = outputRegex.stream().map(reg -> Pattern.compile(reg)).collect(Collectors.toList());
+        }
+    }
+
+    private void fillStopWordsToMap(@NonNull Guardrail guardrail, Map<String, List<String>> map) {
+        List<StopWords> stopWords = guardrail.getStopWords();
+        if (stopWords == null || stopWords.isEmpty()) {
+            return;
+        }
+        for (StopWords e : stopWords) {
+            map.put(e.getIndex(), Arrays.asList(e.getSourceFields()));
+        }
+    }
+
+    public Boolean validate(String input, int type) {
+        switch (type) {
+            case 0: // validate input
+                return validateRegexList(input, inputRegexPattern) && validateStopWords(input, stopWordsIndicesInput);
+            case 1: // validate output
+                return validateRegexList(input, outputRegexPattern) && validateStopWords(input, stopWordsIndicesOutput);
+            default:
+                throw new IllegalArgumentException("Unsupported type to validate for guardrails.");
+        }
+    }
+
+    public Boolean validateRegexList(String input, List<Pattern> regexPatterns) {
+        for (Pattern pattern : regexPatterns) {
+            if (!validateRegex(input, pattern)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public Boolean validateRegex(String input, Pattern pattern) {
+        Matcher matcher = pattern.matcher(input);
+        return !matcher.matches();
+    }
+
+    public Boolean validateStopWords(String input, Map<String, List<String>> stopWordsIndices) {
+        for (Map.Entry entry : stopWordsIndices.entrySet()) {
+            if (!validateStopWordsSingleIndex(input, (String) entry.getKey(), (List<String>) entry.getValue())) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public Boolean validateStopWordsSingleIndex(String input, String indexName, List<String> fieldNames) {
```

Comment:
These function should be a part of that specific guardrail model class and we just call a validate function on the different guardrail component and that will determine how to correctly validate the input.

Reply:
Good suggestion, will refactor it in next pr.